src/Security/Authentication/JwtBearer/src/JwtBearerConfigureOptions.cs

```diff
@@ -39,26 +40,56 @@ public void Configure(string? name, JwtBearerOptions options)
             return;
         }
 
-        var issuer = configSection["ClaimsIssuer"];
-        var audiences = configSection.GetSection("Audiences").GetChildren().Select(aud => aud.Value).ToArray();
+        var issuer = configSection[nameof(TokenValidationParameters.ValidIssuer)];
+        var issuers = configSection.GetSection(nameof(TokenValidationParameters.ValidIssuers)).GetChildren().Select(iss => iss.Value).ToList();
+        if (issuer is not null)
+        {
+            issuers.Add(issuer);
+        }
+        var audience = configSection[nameof(TokenValidationParameters.ValidAudience)];
+        var audiences = configSection.GetSection(nameof(TokenValidationParameters.ValidAudiences)).GetChildren().Select(aud => aud.Value).ToList();
+        if (audience is not null)
+        {
+            audiences.Add(audience);
+        }
+
+        options.Authority = configSection[nameof(options.Authority)] ?? options.Authority;
+        options.BackchannelTimeout = StringHelpers.ParseValueOrDefault(configSection[nameof(options.BackchannelTimeout)], _invariantTimeSpanParse, options.BackchannelTimeout);
+        options.Challenge = configSection[nameof(options.Challenge)] ?? options.Challenge;
```

Comment:
@captainsafia @Tratcher Could it be useful that `EventsType` be set from config too?
```csharp
options.EventsType = StringHelpers.ParseValueOrDefault(configSection[nameof(options.EventsType)], Type.GetType, null);
```

Reply:
Would you really switch out the events dynamically?

I think we've added more than we should have already and we're past the point of diminishing returns.